src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,105 @@
-import { useState } from 'react';
+import { FormEvent, useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({
+  onAdd,
+}) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [newMovieTitle, setNewMovieTitle] = useState('');
+  const [newMovieDescription, setNewMovieDescription] = useState('');
+  const [newMovieImgUrl, setNewMovieImgUrl] = useState('');
+  const [newMovieImdbUrl, setNewMovieImdbUrl] = useState('');
+  const [newMovieImdbId, setNewMovieImdbId] = useState('');
+
+  const isSubmitActive = newMovieTitle !== ''
+    && newMovieImgUrl !== ''
+    && newMovieImdbUrl !== ''
+    && newMovieImdbId !== '';
+
+  const submitHandler = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setCount(prevCount => prevCount + 1);
+
+    const newMovie: Movie = {
+      title: newMovieTitle,
+      description: newMovieDescription,
+      imgUrl: newMovieImgUrl,
+      imdbUrl: newMovieImdbUrl,
+      imdbId: newMovieImdbId,
+    };
+
+    onAdd(newMovie);
+
+    setNewMovieTitle('');
+    setNewMovieDescription('');
+    setNewMovieImgUrl('');
+    setNewMovieImdbUrl('');
+    setNewMovieImdbId('');
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => submitHandler(event)}
```

Comment:
You can omit the wrapper if you still use event as argument.
```suggestion
      onSubmit={submitHandler}
```